### PR TITLE
Document field move scripts, use FLDEFF constants for dofieldeffect

### PR DESCRIPTION
**Pull request by Fontbane**

### CODE REVIEW

File: data/maps/LilycoveCity/scripts.inc [hunk 1]
```diff
@@ -391,7 +391,7 @@ LilycoveCity_EventScript_1E3006:: @ 81E3006
 	waitmovement 0
 	delay 50
 	setfieldeffectargument 0, 1
-	dofieldeffect 30
+	dofieldeffect FLDEFF_NPCFLY_OUT
 	delay 15
 	removeobject VAR_LAST_TALKED
 	waitfieldeffect 30
```
Comment (huderlem) [inline data/maps/LilycoveCity/scripts.inc]: Do the same for `waitfieldeffect`

File: data/scripts/field_move_scripts.inc [hunk 2]
```diff
@@ -1,332 +1,332 @@
...
+Route103_EventScript_CutTree:: @ 82906BB
+Route104_EventScript_CutTree:: @ 82906BB
+Route110_TrickHousePuzzle1_EventScript_CutTree:: @ 82906BB
+Route111_EventScript_CutTree:: @ 82906BB
+Route116_EventScript_CutTree:: @ 82906BB
+Route117_EventScript_CutTree:: @ 82906BB
+Route118_EventScript_CutTree:: @ 82906BB
+Route120_EventScript_CutTree:: @ 82906BB
+Route121_EventScript_CutTree:: @ 82906BB
+Route123_EventScript_CutTree:: @ 82906BB
```
Comment (huderlem) [inline data/scripts/field_move_scripts.inc]: These should all be collapsed into a single `EventScript_CutTree` label.

File: data/scripts/field_move_scripts.inc [hunk 3]
```diff
@@ -1,332 +1,332 @@
...
 	lockall
-	goto_if_unset FLAG_BADGE01_GET, Route103_EventScript_290721
+	goto_if_unset FLAG_BADGE01_GET, Route103_EventScript_CheckTreeCantCut
```
Comment (huderlem) [inline data/scripts/field_move_scripts.inc]: These scripts aren't specific to `Route103`. The same is true for most of the other changes in this file.

**Comment by PikalaxALT** [comment]
Please resolve this linker error:
```
arm-none-eabi-ld: data/event_scripts.o: in function `EventScript_275A50':
(script_data+0x9a3f9): undefined reference to `Route103_Text_290771'
arm-none-eabi-ld: data/event_scripts.o: in function `EventScript_275AA9':
(script_data+0x9a452): undefined reference to `Route103_Text_290771'
arm-none-eabi-ld: data/event_scripts.o: in function `EventScript_275B02':
(script_data+0x9a4ab): undefined reference to `Route103_Text_290771'
arm-none-eabi-ld: data/event_scripts.o: in function `EventScript_275BE8':
(script_data+0x9a5d2): undefined reference to `Route103_Text_290771'
make: *** [pokeemerald.elf] Error 1
```

**Pull request merged**